Sources/NIOCore/NIODecodedAsyncSequence.swift

```diff
@@ -220,3 +303,89 @@ extension NIODecodedAsyncSequence: Sendable where Base: Sendable, Decoder: Senda
 
 @available(*, unavailable)
 extension NIODecodedAsyncSequence.AsyncIterator: Sendable {}
+
+// MARK: - NIOSplitMessageDecoder
+
+/// A decoder which splits the data into subsequences that are separated by a given separator.
+/// Similar to standard library's `String.split(separator:maxSplits:omittingEmptySubsequences:)`.
+///
+/// Use `AsyncSequence/split(omittingEmptySubsequences:maximumBufferSize:whereSeparator:)`
+/// or `AsyncSequence/split(separator:omittingEmptySubsequences:maximumBufferSize:)` to create a
+/// `NIODecodedAsyncSequence` that uses this decoder.
+public struct NIOSplitMessageDecoder: NIOSingleStepByteToMessageDecoder {
```

Comment:
Can this go in a different file? It's unrelated to the `NIODecodedAsyncSequence`.